plugins/woocommerce/tests/legacy/unit-tests/order/class-wc-tests-crud-orders.php

```diff
@@ -611,6 +611,190 @@ public function test_calculate_shipping() {
+		$product_1->save();
+
+		$product_2 = WC_Helper_Product::create_simple_product( false, array( 'name' => 'Test Shipping Product 2', 'sku' => 'TSSKU2' ) );
+		$product_2->set_shipping_class_id( $shipping_class_2['term_id'] );
+		$product_2->save();
+
+		$product_3 = WC_Helper_Product::create_simple_product( false, array( 'name' => 'Test Shipping Product 3', 'sku' => 'TSSKU3' ) );
+		$product_3->set_shipping_class_id( $shipping_class_3['term_id'] );
+		$product_3->save();
+
+		$product_4 = WC_Helper_Product::create_simple_product( false, array( 'name' => 'Test Shipping Product 4', 'sku' => 'TSSKU4' ) );
+		$product_4->set_shipping_class_id( $shipping_class_4['term_id'] );
+		$product_4->save();
+
+		// Create 3 shipping zones. 2 will have one flat rate each while the last will have 2 flat rates.
+		$zone_1 = new WC_Shipping_Zone();
+		$zone_1->set_zone_name( 'New York' );
+		$zone_1->set_zone_order( 1 );
+		$zone_1->add_location( 'US:NY', 'state' );
+		$instance_id_1 = $zone_1->add_shipping_method( 'flat_rate' );
+		$zone_1->save();
+
+		$zone_2 = new WC_Shipping_Zone();
+		$zone_2->set_zone_name( 'Florida' );
+		$zone_2->set_zone_order( 2 );
+		$zone_2->add_location( 'US:FL', 'state' );
+		$instance_id_2 = $zone_2->add_shipping_method( 'flat_rate' );
+		$zone_2->save();
+
+		$zone_3 = new WC_Shipping_Zone();
+		$zone_3->set_zone_name( 'California and Illinois' );
+		$zone_3->set_zone_order( 3 );
+		$zone_3->add_location( 'US:CA', 'state' );
+		$zone_3->add_location( 'US:IL', 'state' );
+		$instance_id_3 = $zone_3->add_shipping_method( 'flat_rate' );
+		$instance_id_4 = $zone_3->add_shipping_method( 'flat_rate' );
+		$zone_3->save();
+
+		$map = array(
+			$shipping_class_1['term_id'] => $instance_id_1, // NY
+			$shipping_class_2['term_id'] => $instance_id_2, // FL
+			$shipping_class_3['term_id'] => $instance_id_3, // CA
+			$shipping_class_4['term_id'] => $instance_id_4, // IL
+		);
+
+		// 2 every flat rate add no cost by default and configure the cost for only one of the shipping classes. Different to each flat rate.
+		foreach ( $map as $shipping_class_id => $instance_id ) {
+			$data = array(
+				'woocommerce_flat_rate_title'                                      => 'Flat rate',
+				'woocommerce_flat_rate_tax_status'                                 => 'taxable',
+				'woocommerce_flat_rate_cost'                                       => '',
+				'woocommerce_flat_rate_class_cost_' . $shipping_class_1['term_id'] => '',
+				'woocommerce_flat_rate_class_cost_' . $shipping_class_2['term_id'] => '',
+				'woocommerce_flat_rate_class_cost_' . $shipping_class_3['term_id'] => '',
+				'woocommerce_flat_rate_class_cost_' . $shipping_class_4['term_id'] => '',
+				'woocommerce_flat_rate_type'                                       => 'class',
+				'instance_id'                                                      => $instance_id,
+			);
+
+			$data['woocommerce_flat_rate_class_cost_' . $shipping_class_id ] = $shipping_class_id * 10;
```

Comment:
(2/3) Then, here, we could change this loop to something like this:

```php
foreach ( $map as $shipping_class_id => $method ) {
	list( $method_id, $method_cost ) = $method;

	$data = array(
		/* As currently defined... */
		'instance_id' => $method_id,
	);

	$data['woocommerce_flat_rate_class_cost_' . $shipping_class_id ] = $method_cost;
	/* Rest of loop... */
}
```

Reply:
Done as well!